### PR TITLE
Include extra questions in projects csv export

### PR DESCRIPTION
**Pull request by gr-eg**

### CODE REVIEW

File: server/routes/handlers/projects.js [hunk 1]
```diff
@@ -146,6 +146,20 @@ module.exports = {
             { label: 'Supervisor Phone', value: 'supervisor.phone' },
           ];
 
+          const answers = data.map(x => x.answers || {});
+          const answerKeys = [
+            ...new Set((answers || []).reduce((acc, val) => {
```
Comment (Wardormeur) [inline server/routes/handlers/projects.js]: You could default the value of answers when it's actually defined rather than patching it on usage

**Comment by Wardormeur** [comment]
Thoughts @gr-eg @DanielBrierton  ?

File: server/routes/handlers/projects.js (hunk 1 above)
Comment (DanielBrierton) [reply server/routes/handlers/projects.js]: Or just load the event and grab the array of answer keys from there.

Comment (Wardormeur) [reply server/routes/handlers/projects.js]: which should available in req.app.locals.event

**Comment by DanielBrierton** [comment]
I think the integration tests aren't failing because the created projects don't have any questions answered, and so when the code loops over all the answers of all the projects, you end up with an empty array of keys, and so no extra columns. Definitely need to update the tests to cover this.

On the `toCSV` part, I think it would definitely be good to split it out. The model doesn't need to know the collection for that, does it? It needs to know about its members, but not other projects.

**Comment by gr-eg** [comment]
Thanks for the feedback. I have modified the project toCSV to grab the question fields from the event rather than the submitted project.

I tried to get the integration tests to assert the CSV response as well but since it is sqlite and the questions field is JSONB the questions column is just null in the tests. I'll add a comment on the Trello card so this doesn't get lost.

I agree the `toCsv` is getting quite long. I'd appreciate a little more guidance on how you think this should be done before I jump in. Happy to pair if that's easier /cc @Wardormeur @DanielBrierton

**Comment by Wardormeur** [comment]
Merging as it, and adding a ticket to externalize later on

**Pull request merged**